lib/schemas/payment.js

```diff
@@ -0,0 +1,93 @@
+const {config} = require('bedrock');
+const {PaymentStatus} = require('bedrock-payment');
+
+const w3cDateTime = require('bedrock-validation/schemas/w3cDateTime');
+const numericFloat = '^[0-9]+(\\.[0-9]+)?$';
+const paymentEnum = Object.values(PaymentStatus);
+// this is used by the validator to ensure you
+// can not post a service not supported by the current project.
+const servicesEnum = config.bedrock_payment.services || [];
+
+const _payment = ({required}) => ({
+  title: 'Bedrock Payment',
+  type: 'object',
+  required,
+  additionalProperties: false,
+  properties: {
+    id: {
+      title: 'id',
+      type: 'string',
+    },
+    amount: {
+      title: 'amount',
+      type: 'string',
+      pattern: numericFloat
+    },
+    currency: {
+      title: 'currency',
+      type: 'string'
```

Comment:
Add `enum` here.

Reply:
Ok I used this: https://github.com/datasets/currency-codes/blob/master/data/codes-all.csv
to make a json array of all the currency codes and added it an enum. Good idea. Was a bit of a new feature and might have been better as a feature request, but was added here: https://github.com/digitalbazaar/bedrock-payment-http/pull/1/commits/efae44a581e765338c880ee2f6b5e8d09f8cd39c

Reply:
We're accepting all these currencies by way of paypal?

Reply:
no we are definitely not accepting all of these currencies, but this validator is for Payments in general not for PayPal specifically.

Reply:
OK, at what point will we be validating against the list of currencies that we'll actually be accepting?

Reply:
@mattcollier Probably in the PayPal layer. I made an issue here: https://github.com/digitalbazaar/bedrock-payment-paypal/issues/1

Stripe did not accept Thai baht last time I checked, but accepted currencies is definitely something to be checked in the Payment plugin. All we need to do here is make sure it's a valid currency at all.

Reply:
@mattcollier I believe we are happy with the validation here and in bedrock-payment-paypal correct?